src/redux/slices/keyring.js

```diff
@@ -247,8 +257,9 @@ export const getICNSData = createAsyncThunk(
 
       // Set the updated currentWallet and wallets.
       const response = await keyring.instance?.getState();
-      const { wallets } = response || {};
-      const wallet = wallets[keyring.currentWallet?.walletNumber];
+      const { wallets, currentWalletId } = response || {};
+
+      const wallet = wallets[currentWalletId];
```

Comment:
if `{ wallets, currentWalletId }` is undefined, we shouldn't execute the next actions right?

`const wallet = wallets[currentWalletId];` this will crash in that case i think